### PR TITLE
Go: Add Log Injection query (CWE-117)

### PR DESCRIPTION
**Pull request by atorralba**
This PR adds a query for Log Injection (CWE-117).

## Description

If unsanitized user input is written to a log entry, a malicious user may be able to forge new log entries.

Forgery can occur if a user provides some input with characters that are interpreted when the log output is displayed. If the log is displayed as a plain text file, then new line characters can be used by a malicious user. If the log is displayed as HTML, then arbitrary HTML may be include to spoof log entries.

## To consider

No sanitizers were added, following the same [reasoning](https://github.com/github/codeql/pull/7054#:~:text=Java%20%3A%20Add%20Log%20Injection%20Vulnerability%C2%A0%235099%20(comment)) as in the Java version.

## Evaluation

The query shows results in 4820 projects. A small sample of them were reviewed and deemed TPs.

## References

https://www.owasp.org/index.php/Log_Injection

### CODE REVIEW

**Comment by owen-mc** [comment]
It looks good to me. Can you explain what's going on with the stubs?

**Comment by atorralba** [comment]
> It looks good to me. Can you explain what's going on with the stubs?

First time using `depstubber`, sorry. I somehow pushed the generated `stub.go` _and_ the actual, downloaded source code of each dependency 😅

**Review by owen-mc — APPROVED** [review]

File: ql/src/Security/CWE-117/LogInjection.qhelp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "qhelp.dtd">
+<qhelp>
+<overview>
+<p>If unsanitized user input is written to a log entry, a malicious user may
+be able to forge new log entries.</p>
+
+<p>Forgery can occur if a user provides some input with characters that are interpreted
+when the log output is displayed. If the log is displayed as a plain text file, then new
+line characters can be used by a malicious user. If the log is displayed as HTML, then
+arbitrary HTML may be include to spoof log entries.</p>
```
Comment (mattpollard) [inline ql/src/Security/CWE-117/LogInjection.qhelp:12]:
```suggestion
arbitrary HTML may be included to spoof log entries.</p>
```

**Review by mattpollard — APPROVED** [review]
I noticed a minor typo, but otherwise, this is good to go 🚀

**Pull request merged**